### PR TITLE
Extend flavor with priority to control service downgrade

### PR DESCRIPTION
**Pull request by baixinsui**
fixes https://github.com/eclipse-xpanse/xpanse/issues/1577
Modify service flavor from lower priority to higher priority successfully:
msedge_KMVyD9sqXa
Modify service flavor from higher priority to lower priority failed:
msedge_CofYZ8ApMV

### CODE REVIEW

File: modules/models/src/main/java/org/eclipse/xpanse/modules/models/servicetemplate/ServiceFlavor.java [hunk 1]
```diff
@@ -34,4 +34,7 @@ public class ServiceFlavor implements Serializable {
     @Schema(description = "The properties of the flavor")
     private Map<String, String> properties;
 
+    @NotNull
+    @Schema(description = "The priority of the flavor. The larger value means higher priority.")
```
Comment (swaroopar) [inline modules/models/src/main/java/org/eclipse/xpanse/modules/models/servicetemplate/ServiceFlavor.java:38]: it must be other way round. "The larger the value means lower priority"

File: modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/DeployService.java [hunk 2]
```diff
@@ -312,6 +325,20 @@ public DeployTask getModifyTask(ModifyRequest modifyRequest,
         return modifyTask;
     }
 
+    private void validateFlavorDowngradedIsAllowed(String originalFlavor, String newFlavor,
+                                                   FlavorsWithPrice flavors) {
+        if (!flavors.isDowngradeAllowed()) {
+            ServiceFlavor newServiceFlavor = getServiceFlavorWithName(newFlavor, flavors);
+            ServiceFlavor originalServiceFlavor = getServiceFlavorWithName(originalFlavor, flavors);
+            if (newServiceFlavor.getPriority() < originalServiceFlavor.getPriority()) {
```
Comment (swaroopar) [inline modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/DeployService.java:333]: should be opposite.

File: modules/models/src/main/java/org/eclipse/xpanse/modules/models/servicetemplate/ServiceFlavor.java [hunk 3]
```diff
@@ -34,4 +34,7 @@ public class ServiceFlavor implements Serializable {
...
+    private int priority;
```
Comment (swaroopar) [inline modules/models/src/main/java/org/eclipse/xpanse/modules/models/servicetemplate/ServiceFlavor.java:39]: would be good if we can have a check that priority minimum allowed value is "1"

File: samples/huawei-Kafka-autofill.yml [hunk 4]
```diff
@@ -103,6 +105,7 @@ flavors:
   modificationImpact:
     isDataLost: true
     isServiceInterrupted: true
+  isDowngradeAllowed: false
```
Comment (swaroopar) [inline samples/huawei-Kafka-autofill.yml:108]: default value would be true. So it is better to have maximum examples with "true" value.

**Review by swaroopar — CHANGES_REQUESTED** [review]

File: modules/models/src/main/java/org/eclipse/xpanse/modules/models/servicetemplate/ServiceFlavor.java (hunk 1 above)
Comment (baixinsui) [reply modules/models/src/main/java/org/eclipse/xpanse/modules/models/servicetemplate/ServiceFlavor.java]: Updated.

File: modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/DeployService.java (hunk 2 above)
Comment (baixinsui) [reply modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/DeployService.java]: Updated.

File: modules/models/src/main/java/org/eclipse/xpanse/modules/models/servicetemplate/ServiceFlavor.java (hunk 3 above)
Comment (baixinsui) [reply modules/models/src/main/java/org/eclipse/xpanse/modules/models/servicetemplate/ServiceFlavor.java:41]: Updated.

**Review by swaroopar — APPROVED** [review]
LGTM

**Pull request merged**